runtime/lua/vim/filetype.lua

```diff
@@ -1693,6 +1690,9 @@ local pattern = {
   ['.*/%.?cmus/.*%.theme'] = 'cmusrc',
   ['.*/%.cmus/autosave'] = 'cmusrc',
   ['.*/%.cmus/command%-history'] = 'cmusrc',
+  ['/etc/hostname%..*'] = starsetf('config'),
```

Comment:
```suggestion
  ['.*/etc/hostname%..*'] = starsetf('config'),
```
? (at least that passes the corresponding test for me.)

Reply:
But that probably means the `match_pattern` function has an issue with its handling of `/` (which also relates to the `file.git/modules//config` failure).

Reply:
That is interesting. Can you make sure you have pushed all your changes? Otherwise it'd be worrying that the tests behave differently on different platforms...

Reply:
Definitely fails for me...

Reply:
Yes, I pushed everything except the `finish` at the start of `filetype.vim`. 

> Otherwise it'd be worrying that the tests behave differently on different platforms...

I am on Linux Mint btw.

Reply:
Ah, I think I see what's going on: this is indeed platform-specific: macOS sandboxes the `/etc` files; trying to edit `/etc/foo` will open `/private/etc/foo`. Hence Vim's pattern here will not work (and Vim in fact behaves the same). 

That means we have three options:
1. accept the failure (`/etc/hostname.*` is not relevant on macOS)
2. fix the failure (all other patterns are `*/etc/<foo>`, so this might be an oversight?)
3. report the issue upstream and see what they thing (@zeertzjq ?)

Reply:
> report the issue upstream and see what they think

We should definitely do this one, maybe Vim will simply update the pattern to `*/etc/hostname.*` which we can then update as well

Reply:
I'm quite sure it's an oversight, though -- _every single_ other of the >100 patterns uses `*/etc`.

Reply:
Forgot to put a `finish` in `scripts.vim`, anything else? Because everything is failing right now.

Reply:
Test fails on macOS CI as well; not sure yet why this is not an issue for `filetype.vim`, but that means we need to adapt the pattern to have the same behavior, independent of what Vim does.

Reply:
Ok, will do that